### DISCUSSION TITLE
We want your feedback on the NetBox Labs plugin certification program

### DISCUSSION THREAD

**Discussion by jeffgdotorg** [netbox-community/netbox#15767] [General]
Hey NetBox fans, I've just posted the first public [writeup](https://github.com/netbox-community/netbox/wiki/Plugin-Certification-Program) detailing the goals and mechanics of the plugin certification program announced in March. I would love to hear your feedback, whether you're a plugin user or a plugin maker. Thanks in advance!

**Comment by amuckart** [comment]
I am not a dev, but as a user and maintainer of Netbox I think this is an _excellent_ idea. 

Knowing that I can install a plugin and have less worry about it being abandoned, breaking upgrades, etc. will make us a lot more likely to use them.

**Comment by netravnen** [comment]
I see it as a perfect upsell to the hosted Netbox solution, Netbox Labs is offering. Where the certified plugins are guaranteed (by Netbox Labs) to work with the current Netbox version. And can be enabled by the customers of the hosted Netbox solution.

From a self-hosted perspective. The valued add to Netbox Labs (for the certified plugins Netbox Labs maintains) might be less obvious, IMHO, as there is no direct stable revenue stream back to Netbox Labs from the OSS users. (💰 = Dev-time)

For Netbox Certified plugins, offered by 3rd-party entities, I think it is a great addition. Making which plugins can be expected to work with (at minimum) current and current-1 feature release(s) more obvious to all Netbox users. As long as the 3rd-party has a minimum level of commitment of wanting to maintain the Netbox Certified plugin are offering.

> For lead maintainers of certified plugins:
> - Offer of repo transfer to netbox-community

I find this especially valuable. To (try and) avoid a repeat of the situation with the `netbox-dns` plugin. That needed to be forked to `netbox-plugin-dns`. With all the ensuing confusion of which repo is the active one. (This one disappears the more breaking changes ensures the old one no longer works)

> Knowing that I can install a plugin and have less worry about it being abandoned, breaking upgrades, etc. will make us a lot more likely to use them.

I wholeheartedly agree with this statement as a Netbox OSS user.

> Version compatibility metadata
> Each release of a certified plugin must include metadata indicating which versions of NetBox are compatible with the plugin release. Currently this metadata is human-readable and lives in the plugin's top-level README. A schema for expressing this metadata in machine-readable form will be introduced once it has stabilized.

From the announcement. This I find valuable in the long term. Instead of needing to manually add (and later remove) version restrictions to the `local_requirements.txt` file. When a new plugin version that is automatically installed during `./upgrade.sh` run is not compatible with current-1 version of Netbox. Thereby needing to hold back on upgrading Netbox and plugins. Until after all locally installed plugins have reached Netbox >= 4.0 combability.
